cypress/integration/carry_okay_spec.js

```diff
@@ -93,94 +106,96 @@ describe('CarryOkay', () => {
-        .and("contain","Hip Hop")
-        .and("contain","R&B")
+      cy.get('[data-cy=song-card]').should('contain', 'My Boo')
+        .and('contain', 'Usher')
+        .and('contain','Hip Hop')
+        .and('contain','R&B');
       cy.get('[data-cy=song-card-btn]').click();
       cy.get('[data-cy=song-card]').should('not.exist');
     });
-
   });
 
   describe('Song Book', () => {
 
-  beforeEach(() => {
-    cy.intercept( 'https://carryokay-server.herokuapp.com/genres', {fixture:"genre_data.json"})
-    cy.intercept('https://carryokay-server.herokuapp.com/songs', {fixture:"song_data.json"})
-    cy.visit('http://localhost:3000');
-    cy.get('[data-cy=song-book-nav]').click();
-    });
-
-  it('should display Song Book page components', () => {
-    cy.get('[data-cy=home-nav]').should('exist');
-    cy.get('[data-cy=my-songs-nav]').contains("My Songs");
-    cy.get('[data-cy=song-book-nav]').contains("Song Book");
-    cy.get('[data-cy=search-form]').should('exist');
-    cy.get('[data-cy=song-card]').should("have.length", 2);
-    cy.get('[data-cy=song-card]').eq(0).contains("Radiohead");
-    cy.get('[data-cy=song-card]').eq(1).contains("Usher");
-    cy.get('[data-cy=song-card]').first().should("contain", "Jigsaw falling into place")
-    .and("contain", "Radiohead")
-    .and("contain","Electronica")
-    .and("contain","Rock")
-  cy.get('[data-cy=album-img]').first().should('exist');
-  cy.get('[data-cy=song-card-btn]').first().should('exist');
-  cy.get('[data-cy=microphone-icon]').should('exist');
-  });
+    beforeEach(() => {
+      cy.get('[data-cy=song-book-nav]').click();
+      });
 
-  it('should allow user to make a search', () => {
-    cy.get('[data-cy=search-bar]').type('usher');
-    cy.get('[data-cy=search-btn').click()
-    cy.get('[data-cy=song-card]').should("have.length", 1)
-      .and('contain', 'Usher');
-    cy.get('[data-cy=search-bar]').clear()
-    cy.get('[data-cy=search-bar]').type('body')
-    cy.get('[data-cy=search-btn').click()
-    cy.get('[data-cy=song-card]').should('have.length', 2);
-  });
+    it('should display Song Book page components', () => {
+      cy.get('[data-cy=home-nav]').should('exist');
+      cy.get('[data-cy=my-songs-nav]').contains('My Songs');
+      cy.get('[data-cy=song-book-nav]').contains('Song Book');
+      cy.get('[data-cy=search-form]').should('exist');
+      cy.get('[data-cy=song-card]').should('have.length', 2);
+       
+      cy.fixture('/song_data.json').then((data) => {
+        data.forEach(song => {
+          cy.get('[data-cy=song-card]').contains(song.title);
+          cy.get('[data-cy=song-card]').contains(song.artist);
+          });
+        });
+      cy.fixture('/genre_data.json').then((data) => {
+        data.forEach(genre => {
+          const genres = Object.entries(genre)
+          genres.forEach(entry => {
+            if(entry.includes(true)) {
+              cy.get('[data-cy=song-card]').contains(entry[0]);
+            }
+          });
```

Comment:
these fixtures looping through with a forEach essentially check every entry, perhaps the first more than the second but potentially shorten it for optimization if a preference

Reply:
What do you mean by shorten it?

Reply:
instead of a ForEach maybe a while or do/while loop limiting it to 305 iterations

Reply:
ok, we can look at it together, I don't have much experience with the do..while loop